lib/localparticipant.js

```diff
@@ -222,15 +191,15 @@ class LocalParticipant extends Participant {
         if (error) {
           trackSignaling.removeListener('updated', updated);
           log.warn(`Failed to publish the ${util.trackClass(localTrack, true)}: ${error.message}`);
-          self._removeTrack(localTrack);
+          self._removeTrack(localTrack, localTrack.id);
```

Comment:
@markandrus Because `Participant#_addTrack/_removeTrack` needs to work for both `LocalTrack`s, which have `.id` property, and `RemoteTrack` whose `ID` needs to be derived from its corresponding `RemoteTrackPublicationSignaling` instance.